### PR TITLE
Implement DLTClient for UDP multicast

### PR DESCRIPTION
**Pull request by joaodasousa**
1 - change line:
        #define MULTICAST_GROUP_ADDRESS "239.1.2.3" on file dlt-example-multicast-client.c
2 - Compile and run dlt-example-multicast-client example application

### CODE REVIEW

**Comment by ssugiura** [comment]
@joaodasousa Apart from comments in the changes, I would like to ask you two things:
- Please explain what was the problem and how you solved/improved it so that everyone can understand what you fixed with this commit.
- Please add signed-off
Thank you!

File: src/examples/CMakeLists.txt [hunk 1]
```diff
@@ -42,9 +42,13 @@ if(WITH_UDP_CONNECTION)
...
 			RUNTIME DESTINATION bin
 			COMPONENT base)
-
 endif(WITH_UDP_CONNECTION)
 
-install(TARGETS dlt-example-user dlt-example-user-func dlt-example-user-common-api dlt-example-filetransfer
+set(dlt_example_multicast_client_SRCS dlt-example-multicast-client.c)
+add_executable(dlt-example-multicast-client ${dlt_example_multicast_client_SRCS})
+target_link_libraries(dlt-example-multicast-client dlt)
+set_target_properties(dlt-example-multicast-client PROPERTIES LINKER_LANGUAGE C)
```
Comment (ssugiura) [inline src/examples/CMakeLists.txt:50]: Add these lines inside `if(WITH_UDP_CONNECTION)`.

File: src/examples/dlt-example-multicast-client.c [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#include <stdio.h>
+#include <syslog.h>
+#include <time.h>
+#include <sys/time.h>
+
+#include "dlt_client.h"
+#include "dlt_client_cfg.h"
+#include "dlt_types.h"
+
+#define MULTICAST_GROUP_ADDRESS "239.1.2.3"
```
Comment (ssugiura) [inline src/examples/dlt-example-multicast-client.c:33]: Can we align this address to default defined in `dlt.conf`?

File: src/lib/dlt_client.c [hunk 3]
```diff
@@ -153,16 +154,17 @@ DltReturnValue dlt_client_init(DltClient *client, int verbose)
         dlt_vlog(LOG_INFO,
                  "Init dlt client struct with default port: %hu.\n",
                  servPort);
-
     return dlt_client_init_port(client, servPort, verbose);
 }
 
 DltReturnValue dlt_client_connect(DltClient *client, int verbose)
 {
+    const int yes = 1;
```
Comment (ssugiura) [inline src/lib/dlt_client.c:162]: As this value is used only if DLT_CLIENT_MODE_UDP_MULTICAST, I propose to define this line inside there. Same for line 167

File: src/shared/dlt_common.c [hunk 4]
```diff
@@ -2035,12 +2037,23 @@ int dlt_receiver_receive(DltReceiver *receiver, DltReceiverType from_src)
...
+            /* wait for data from UDP socket */
+            addrlen = sizeof(receiver->addr);
+            receiver->bytesRcvd = recvfrom(receiver->fd,
+                                           receiver->buf + receiver->lastBytesRcvd,
+                                           receiver->buffersize - receiver->lastBytesRcvd,
+                                           0,
+                                           (struct sockaddr *)&(receiver->addr),
+                                           &addrlen);
+        }
+
```
Comment (ssugiura) [inline src/shared/dlt_common.c:2056]: Please correct indention here

File: src/examples/CMakeLists.txt (hunk 1 above)
Comment (joaodasousa) [reply src/examples/CMakeLists.txt]: Agree

File: src/examples/dlt-example-multicast-client.c (hunk 2 above)
Comment (joaodasousa) [reply src/examples/dlt-example-multicast-client.c]: Agree

File: src/shared/dlt_common.c (hunk 4 above)
Comment (joaodasousa) [reply src/shared/dlt_common.c]: Agree

File: src/lib/dlt_client.c (hunk 3 above)
Comment (joaodasousa) [reply src/lib/dlt_client.c:162]: I don't agree with this comment.
Defining variables after labels (DLT_CLIENT_MODE_UDP_MULTICAST) is not valid in C. Although workarounds exists, in my opinion they are kinda ugly :)

**Comment by ssugiura** [comment]
@joaodasousa I took a deeper look and have an idea that we can merge the code in src/examples/dlt-example-multicast-client.c to src/console/dlt-receive.c so that we would have one common DLT log viewer client. What do you think?

**Comment by joaodasousa** [comment]
> @joaodasousa I took a deeper look and have an idea that we can merge the code in src/examples/dlt-example-multicast-client.c to src/console/dlt-receive.c so that we would have one common DLT log viewer client. What do you think?

Makes perfect sense. I have done as suggested

**Pull request merged**